### PR TITLE
[REF] Improve Get Coordinates by supporting other geocoding providers…

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This improves the handling in getCoordinates API action to allow for providers other than Google.

Improves on https://github.com/civicrm/civicrm-core/pull/23597

Before
----------------------------------------
Only Google provider supported which not all users use

After
----------------------------------------
Any Provider e.g. Google or [OSM](https://github.com/systopia/de.systopia.osm) can be used for the proximity search if appropriately supported

### CODE REVIEW

**Comment by colemanw** [comment]
@seamuslee001 I think we need to add a unit test for this api action. A simple way would be to add a `getCoordinates` function in `CRM_Utils_Geocode_TestProvider` that always returns the same dummy coords.

**Comment by eileenmcnaughton** [comment]
what is the trick to this - geocoder obviously has a handful of providers (including OSM/ Nominatim)

**Comment by seamuslee001** [comment]
@colemanw I have added a unit test and @eileenmcnaughton the idea is that an address as a string is passed through to a function getCoordinates() on the geocoder provider which the APIv4 expects an array to come back either empty or with geo_code_1 and geo_code_2 set or geo_code_error as the only array key

**Comment by eileenmcnaughton** [comment]
ok - seems not too hard to do

**Comment by colemanw** [comment]
Unit test looks good.

**Comment by eileenmcnaughton** [comment]
Failure relates - teardown maybe?

image

File: settings/Map.setting.php [hunk 1]
```diff
@@ -52,6 +52,9 @@
     'pseudoconstant' => [
       'callback' => 'CRM_Core_SelectValues::geoProvider',
     ],
+    'on_change' => [
+      'CRM_Utils_GeocodeProvider::reset',
+    ],
```
Comment (colemanw) [inline settings/Map.setting.php:57]: This is the magic sauce that was previously missing.

**Pull request merged**